rust/src/binary_view_event_handler.rs

```diff
@@ -0,0 +1,66 @@
+use crate::binaryview::BinaryView;
+use binaryninjacore_sys::{
+    BNBinaryView, BNBinaryViewEventType, BNNewViewReference, BNRegisterBinaryViewEvent,
+};
+
+pub trait BinaryViewEventHandler: 'static + Sync {
+    fn on_event(&mut self, binary_view: &BinaryView);
+}
+
+pub type BinaryViewEventType = BNBinaryViewEventType;
+
+/// Registers an event listener for binary view events.
+///
+/// # Example
+///
+/// ```rust
+/// use binaryninja::binaryview::BinaryView;
+/// use binaryninja::binary_view_event_handler::{BinaryViewEventType, BinaryViewEventHandler, register_binary_view_event};
+///
+/// struct EventHandlerContext {
+///     // Context holding state available to event handler
+/// }
+///
+/// impl BinaryViewEventHandler for EventHandlerContext {
+///     fn on_event(&mut self, binary_view: &BinaryView) {
+///         // handle event
+///     }
+/// }
+///
+/// #[no_mangle]
+/// pub extern "C" fn CorePluginInit() {
+///     let context = EventHandlerContext { };
+///
+///     register_binary_view_event(
+///         BinaryViewEventType::BinaryViewInitialAnalysisCompletionEvent,
+///         context,
+///     );
+/// }
+/// ```
+pub fn register_binary_view_event<Handler>(event_type: BinaryViewEventType, handler: Handler)
+where
+    Handler: BinaryViewEventHandler,
+{
+    unsafe extern "C" fn on_event<Handler: BinaryViewEventHandler>(
```

Comment:
I don't think this needs to be unsafe. You seem to wrap the rest of the unsafe blocks down below, which is the style used in most the API.